ledger/participant-integration-api/src/main/scala/platform/store/backend/common/CommonStorageBackend.scala

```diff
@@ -809,43 +825,106 @@ private[backend] trait CommonStorageBackend[DB_BATCH] extends StorageBackend[DB_
       .asVectorOf(contractStateRowParser)(connection)
 
   // Events
-
-  def pruneEvents(pruneUpToInclusive: Offset)(connection: Connection): Unit = {
+  def pruneEvents(pruneUpToInclusive: Offset, pruneAllDivulgedContracts: Boolean)(
+      connection: Connection
+  ): Unit = {
     import com.daml.platform.store.Conversions.OffsetToStatement
-    List(
+    val conditionalImmediateDivulgencePruning =
+      pruneImmediateDivulgence(pruneUpToInclusive, pruneAllDivulgedContracts)
+
+    val consumingExercisesPruning =
       SQL"""
-          -- Divulgence events (only for contracts archived before the specified offset)
-          delete from participant_events_divulgence delete_events
+          -- Exercise events (consuming)
+          delete from participant_events_consuming_exercise delete_events
+          where
+            delete_events.event_offset <= $pruneUpToInclusive"""
+
+    val nonConsumingExercisesPruning =
```

Comment:
To aid readability: move the value definitions into the order in which they are executed.

Potentially even better: just execute them directly, guarding the ones you don't want with a simple `if` statement. This way a reader can just follow along.

Reply:
Updated according to the discussion had offline.